feat: add page for troubleshooting webcontainers

d3lm opened the pull request:
# PR Description

This PR fixes the issue number: SBPF-2824

### Summary of my changes and explanation of my decisions

Some errors may be confusing including `Cannot load native addon because loading addons is disabled` so I added a new section to the WebContainers docs for troubleshooting.

### Self-check

Please check all that apply:

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my code or content update and edited it to the best of my abilities
- [x] I have commented my code, particularly in hard-to-understand areas; my comments are concise

### Contributors list

Would you like your contribution to be celebrated? Please check all that apply:

- [ ] I would like to be featured on the future contributors list in the README. If so, please tell us:
  - what name you'd like to be shown there?
  - we usually link github profile pic -- is that ok? If not, provide another url: 
  - we usually link your github profile but if you have a portfolio page, provide a link:

- [ ] I would like to get a shoutout in the next monthly updates post. If so, please provide your twitter handle (or we will link to your GitHub profile).

⚡️ ⚡️ ⚡️  Thank you for contributing to StackBlitz ⚡️ ⚡️ ⚡️

d3lm commented:
Added and submitted via Codeflow 🔥

sylwiavargas commented on docs/platform/webcontainers/troubleshooting.md:
```diff
@@ -0,0 +1,13 @@
+
+## Cannot load native addon
+
+Currently, WebContainers can only execute languages that are natively supported on the Web  including JavaScript and WebAssembly. It is not possible to run [native addons](https://nodejs.org/api/addons.html) which are usually implemented using native languages such as C++, unless they can be compiled to WebAssembly. Therefore, loading native addons is disabled by default via [`--no-addons`](https ...
```
```suggestion
Currently, WebContainers can only execute languages that are natively supported on the Web, including JavaScript and WebAssembly. It is not possible to run [native addons](https://nodejs.org/api/addons.html) which are usually implemented using native languages such as C++, unless they can be compiled to WebAssembly. Therefore, loading native addons is disabled by default via [`--no-addons`](https://nodejs.org/api/cli.html#--no-addons) in WebContainers. As a result, you may encounter an error that says: `Cannot load native addon because loading addons is disabled`. 

There is no way to opt-in into loading native addons because they can't be executed.
```

sylwiavargas replied:
I added two interpunction signs and a joining clause.

jrvidal commented on docs/platform/webcontainers/troubleshooting.md:
```diff
@@ -0,0 +1,13 @@
+
+Currently, WebContainers can only execute languages that are natively supported on the Web  including JavaScript and WebAssembly. It is not possible to run [native addons](https://nodejs.org/api/addons.html) which are usually implemented using native languages such as C++, unless they can be compiled to WebAssembly. Therefore, loading native addons is disabled by default via [`--no-addons`](https ...
+
+The solution to this is to use an alternative to the native addon which is either fully JavaScript or can be compiled to WebAssembly.
```
```suggestion
The solution to this is to use an alternative to the native addon which is either fully implemented on JavaScript or can be compiled to WebAssembly.
```

sylwiavargas reviewed (changes requested):
Thank you for this page. So helpful and a good stub to build on.
I added one suggestion. Apart from that, could you change the page title and URL (not the menu item name because it's visible only in "WebContainers" section so it's self-explanatory) to "Troubleshooting WebContainers"? I imagine we will have a few more troubleshooting pages so I don't want anyone to be confused when querying something in Algolia :)

jrvidal commented on docs/data.ts:
```diff
@@ -48,6 +48,7 @@ export const webcontainersLinks = [
   { text: 'Browser configuration', link: '/platform/webcontainers/browser-config' },
   { text: 'Project configuration', link: '/platform/webcontainers/project-config' },
   { text: 'Turbo package manager', link: '/platform/webcontainers/turbo-package-manager' },
+  { text: 'Troubleshooting', link: '/platform/webcontainers/troubleshooting' },
```
Instead of "troubleshooting", I think it'd make more sense to call this "Runtime limitations" or something like that, where we can explain what WebContainers can and cannot do.

For instance, I was just triaging an issue in webcontainer-core about how we can't register service workers in the pages served by localservice.

sylwiavargas replied:
I think this is a good idea but a separate page (or a subpage) - we can link to these sections from there but my assumption is that not that every known limitation will result in a bug. What do you think?

The troubleshooting page offers "problems - solutions" type of content and I think it might be really helpful to have one place where you know that all your troubles will be addressed. What do you think?

d3lm replied:
I honestly don't have a hard opinion here. Troubleshooting is more general whereas "Runtime limitations" is more specific, and I think limitations can be seen as troubleshooting. So whatever works for me.

jrvidal replied:
"Troubleshooting" for me sounds like "FAQ": a bag of varied things for folks that are lost. I'd rather improve the core doc pages so people can search appropriately, and maybe have an FAQ/troubleshooting page that is just a listicle linking to other pages/sections.

:shrug:

d3lm replied:
Ok, what structure do you suggest? Is it `Troubleshooting -> Runtime limitations` or what exactly do you have in mind?

jrvidal reviewed (approved):
:+1: Nice

d3lm commented:
I have addressed your feedback except for the open question whether we want to have "Runtime limitations" or not and how it's structured. Like I said, I don't have a strong opinion here.
